### PR TITLE
Make dashboard logs button work in the jobs UI for driver-less submission jobs

### LINKED ISSUE #28577: [Jobs] [Dashboard] Job Submission logs for entrypoint scripts without ray.init() don't appear in dashboard UI
### What happened + What you expected to happen

In the "Jobs" tab, logs only appear for jobs that involve at least one Ray driver.

Screen Shot 2022-09-16 at 1 35 17 PM

To find the logs for `ray job submit`, you have to go to the "Logs" tab:
Screen Shot 2022-09-16 at 1 35 37 PM

It should be possible to surface this log in the "Jobs" tab.

### Versions / Dependencies

Ray 3.0.0dev0, Python 3.8, Mac OS

### Reproduction script

`ray start --head`

`ray job submit -- echo hello`

```python
import ray
ray.init()
def f():
   print("hi from task")

print("hi from driver")
ray.get(f.remote())
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by alanwguo**
Signed-off-by: Alan Guo <aguo@anyscale.com>

## Why are these changes needed?

When the job is submitted without a Ray driver, we currently don't display the log button. This PR fixes the issue. 

## Related issue number
fixes #28577 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Thanks for the fix! Looks good to me. Might be good to get a review from someone with javascript/typescript experience

File: dashboard/client/src/type/job.d.ts [hunk 1]
```diff
@@ -70,6 +70,7 @@ export type UnifiedJob = {
   metadata: { [key: string]: string } | null;
   runtime_env: { [key: string]: string } | null;
   driver_info: DriverInfo | null;
+  driver_agent_http_address: string | null;
```
Comment (galenhwang) [inline dashboard/client/src/type/job.d.ts:73]: nit:
```suggestion
  driver_agent_http_address: string?;
```

VS code shows this as shorthand for `string | null`, although maybe it's not the case and type? includes undefined too which is why the other fields are defined like this

File: dashboard/client/src/pages/job/JobRow.tsx [hunk 2]
```diff
@@ -56,6 +57,29 @@ export const JobRow = ({
     }
   })();
 
+  const logsLink = (() => {
+    let link: string | undefined = undefined;
```
Comment (galenhwang) [inline dashboard/client/src/pages/job/JobRow.tsx:61]: nit: assignment of undefined is redundant
```suggestion
    let link: string | undefined;
```

File: dashboard/client/src/pages/job/JobRow.tsx [hunk 3]
```diff
@@ -56,6 +57,29 @@ export const JobRow = ({
...
+
+    if (link) {
+      return (
+        <Link to={link} target="_blank">
+          Log
+        </Link>
+      );
+    }
+
+    return "-";
```
Comment (galenhwang) [inline dashboard/client/src/pages/job/JobRow.tsx:80]: nit / personal pref:
```suggestion
    if (driver_agent_http_address) {
      link = `/log/${encodeURIComponent(
        `${driver_agent_http_address}/logs`,
      )}`;
    } else if (driver_info && ipLogMap[driver_info.node_ip_address]) {
      link = `/log/${encodeURIComponent(
        ipLogMap[driver_info.node_ip_address],
      )}`;
    } else {
      return "-";
    }

    link += `?fileName=${type === "DRIVER" ? job_id : `driver-${submission_id}`}`;

    
    return (
      <Link to={link} target="_blank">
        Log
      </Link>
    );
```

File: dashboard/client/src/pages/job/JobRow.tsx [hunk 4]
```diff
@@ -56,6 +57,29 @@ export const JobRow = ({
     }
   })();
 
+  const logsLink = (() => {
+    let link: string | undefined = undefined;
+    if (driver_agent_http_address) {
+      link = `/log/${encodeURIComponent(
+        `${driver_agent_http_address}/logs`,
+      )}?fileName=${type === "DRIVER" ? job_id : `driver-${submission_id}`}`;
+    } else if (driver_info && ipLogMap[driver_info.node_ip_address]) {
```
Comment (galenhwang) [inline dashboard/client/src/pages/job/JobRow.tsx:66]: Are there instances where both cases can be true (both fields are specified)? Wouldn't the two endpoints be the same or show equivalent info?

**Review by galenhwang — APPROVED** [review]

File: dashboard/client/src/pages/job/JobRow.tsx (hunk 4 above)
Comment (alanwguo) [reply dashboard/client/src/pages/job/JobRow.tsx:66]: yes, I assume they will lead to the same place so I just pick one to prioritize.

**Comment by rkooo567** [comment]
can you update the PR descritpion?

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**